server/routes/lists.js

```diff
@@ -0,0 +1,152 @@
+const { Op } = require("sequelize");
+const express = require("express");
+const {
+  validate,
+  listIdCheck,
+  titleCheck,
+} = require("../middlewares/validate");
+const router = express.Router();
+const authMiddleware = require("../middlewares/authMiddleware");
+const db = require("../models");
+const ListProductModel = require("../models/listProductModel");
+const UserList = require("../models/userListModel");
+const ProductModel = require("../models/productModel");
+
+const giveServerError = (res) =>
+  res.status(500).send({ errors: [{ msg: "Server error" }] });
+
+//Returns id of each list belonging to the user.
+const getLists = async (req, res) => {
+  try {
+    const results = await UserList.findAll({
+      attributes: ["id"],
+      where: { user_id: req.user.id },
+    });
+    res.status(200).send(results);
+  } catch (error) {
+    console.error(error);
+    giveServerError(res);
+  }
+};
+
+//Returns product information for a given list belonging to the user.
+const getList = async (req, res) => {
+  //Queries all the products of each list, as well as the link for each of those products, and the user each of those
+  //lists belongs to. Only returns products of the specified list, and only if that list belongs to the user.
+  try {
+    const [results] = await db.query(`
+        SELECT "Products".name, "Products".link, "Products".id FROM "ListProducts" 
+        JOIN "Products" ON ("ListProducts".product_id = "Products".id)
+        JOIN "UserLists" ON ("ListProducts".list_id = "UserLists".id)
+        WHERE ( "ListProducts".list_id = ${parseInt(
+          req.params.listId
+        )} AND "UserLists".user_id = ${req.user.id} )
```

Comment:
this can probably be updated to a sequelize query for easier readability

Reply:
I was having trouble turning this particular query into a working sequelize statement. I *think* I know what's causing it, but I'm not 100% sure. If it's important, I can try working on it some more and see if I can get it working.